private/buf/bufsync/commits_to_sync_test.go

```diff
@@ -66,101 +64,37 @@ func TestCommitsToSyncWithNoPreviousSyncPoints(t *testing.T) {
 		},
 	}
 	for _, withOverride := range []bool{false, true} {
-		mockBSRChecker := newMockSyncGitChecker()
 		for _, tc := range testCases {
 			func(tc testCase) {
 				t.Run(fmt.Sprintf("%s/override_%t", tc.name, withOverride), func(t *testing.T) {
 					const moduleDir = "."
-					moduleDirsToIdentityOverride := make(map[string]bufmoduleref.ModuleIdentity)
-					if withOverride {
-						moduleDirsToIdentityOverride[moduleDir] = moduleIdentityOverride
-					} else {
-						moduleDirsToIdentityOverride[moduleDir] = nil
-					}
-					testSyncer := syncer{
-						repo:                                  repo,
-						storageGitProvider:                    storagegit.NewProvider(repo.Objects()),
-						logger:                                zaptest.NewLogger(t),
-						errorHandler:                          &mockErrorHandler{},
-						modulesDirsToIdentityOverrideForSync:  moduleDirsToIdentityOverride,
-						sortedModulesDirsForSync:              []string{"."},
-						syncAllBranches:                       true,
-						syncedGitCommitChecker:                mockBSRChecker.checkFunc(),
-						commitsToTags:                         make(map[string][]string),
-						modulesDirsToBranchesToIdentities:     make(map[string]map[string]bufmoduleref.ModuleIdentity),
-						modulesToBranchesExpectedSyncPoints:   make(map[string]map[string]string),
-						modulesIdentitiesToCommitsSyncedCache: make(map[string]map[string]struct{}),
+					opts := []bufsync.SyncerOption{
+						bufsync.SyncerWithAllBranches(),
 					}
-					require.NoError(t, testSyncer.prepareSync(context.Background()))
-					var moduleIdentity bufmoduleref.ModuleIdentity
 					if withOverride {
-						moduleIdentity = moduleIdentityOverride
+						opts = append(opts, bufsync.SyncerWithModule(moduleDir, moduleIdentityOverride))
 					} else {
-						moduleIdentity = moduleIdentityInHEAD
+						opts = append(opts, bufsync.SyncerWithModule(moduleDir, nil))
 					}
-					syncableCommits, err := testSyncer.branchSyncableCommits(
-						context.Background(),
-						moduleDir,
-						moduleIdentity,
-						tc.branch,
-						"", // no expected git sync point
+					handler := newMockSyncHandler()
```

Comment:
I'm confused why this test is not broken. The mock BSR sync point checker should be shared across test cases (ie out of the `for _, tc := range testCases` loop) so when syncing main, the other branches will pick up new sync points.

Reply:
This is totally fine. The repo looks like this:
```
| o-o----------o-----------------o (master)
|   └o-o (foo) └o--------o (bar)
|               └o (baz)
```
So `master` syncs 4 commits, then `foo` syncs 2, `bar` syncs 2, and `baz` syncs one.

If we change the order we'll get different counts, of course, but because I'm using `WithAllBranches()`, it's alphabetical order with git default first, so it syncs in that order.

aside: I will be changing this test in my logic refactor to actually explode out pairwise all the branch orders and assert that way. That would more correctly capture all the cases.

Reply:
Technically I only need to sync once, but I'm going to change this soon anyway, so the subsequent calls are all no-ops for now.

Reply:
If you're using `AllBranches` then you actually don't need to wrap this into multiple test cases, that's why it's not failing. For each test cases it's syncing all branches, and each time you're just asserting one branch. If you don't pass the `AllBranches` option and move the `handler := newMockSyncHandler()` declaration up top to where it was, the test should keep working as it was originally intended, which is sync once branch at a time and assert.

Reply:
Yes, but see my comment above, I chose to keep it that way for now. Can change if you feel strongly, but I mainly wanted to minimize changes here to "prove" that nothing is broken.

Reply:
I'll move the syncer down, actually, I'll need that anyway.

Reply:
The way it is written now it's redundant:

- sync all
- assert main
- sync all
- assert foo
- sync all
- assert bar
- sync all
- assert baz

That's why it was confusing to me, so what I suggest is option 1:

- sync main
- assert main
- sync foo
- assert foo
- sync bar
- assert bar
- sync baz
- assert baz

or option 2:

- sync all
- assert main
- assert foo
- assert bar
- assert baz